### PR TITLE
lesson 7 assignment

### PR DESCRIPTION
**Pull request by jayjonesseattle**
I am stuck on lesson 7 still at step 5. please take a look at the code I have and advise me on how to move forward. any help you can give is appreciated. Thank you

### CODE REVIEW

File: students/terrance_jones/lesson7/html_render.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    def append(self, content):
+        raise NotImplementedError
+
+class Title(OneLineTag):
+    tag = "title"
+
+class SelfClosingTag(Element):
+    def render(self, out_file):
+        tag = self._open_tag()[:-1] + " />\n"
+        outfile.write(tag)
```
Comment (natasha-aleksandrova) [inline students/terrance_jones/lesson7/html_render.py]: you should define `append` method and `raise TypeError`

File: students/terrance_jones/lesson7/html_render.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class Hr(SelfClosingTag):
+    tag = "hr"
+
```
Comment (natasha-aleksandrova) [inline students/terrance_jones/lesson7/html_render.py]: also add `Br` class similar to your `Hr` class

**Comment by natasha-aleksandrova** [comment]
@jayjonesseattle I'm going to close this out since you're still working on this.

Please do continue commenting on this PR if you would like have further discussion; happy to help in any way I can to ensure you don't get behind on this assignment (I promise Lesson 8 is easier 🙂 )

**Comment by jayjonesseattle** [comment]
Yea, Im getting failures from the test that Im not sure how to fix.

=================================== FAILURES
===================================

*___________________________________ test_hr
____________________________________*


*    def test_hr():*

*        """a simple horizontal rule with no attributes"""*

*        hr = Hr()*

*>       file_contents = render_result(hr)*


*test_html_render.py*:231:

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_ _

*test_html_render.py*:30: in render_result

*    element.render(outfile)*

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_ _


self = <html_render.Hr object at 0x108400438>

out_file = <_io.StringIO object at 0x10837aee8>


*    def render(self, out_file):*

*>       tag = self._open_tag()[:-1] + " />\n"*

*E       AttributeError: 'Hr' object has no attribute '_open_tag'*


*html_render.py*:66: AttributeError

*_________________________________ test_hr_attr
_________________________________*


*    def test_hr_attr():*

*        """a horizontal rule with an attribute"""*

*        hr = Hr(width=400)*

*>       file_contents = render_result(hr)*


*test_html_render.py*:239:

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_ _

*test_html_render.py*:30: in render_result

*    element.render(outfile)*

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
_ _


self = <html_render.Hr object at 0x10855dc50>

out_file = <_io.StringIO object at 0x10852b9d8>


*    def render(self, out_file):*

*>       tag = self._open_tag()[:-1] + " />\n"*

*E       AttributeError: 'Hr' object has no attribute '_open_tag'*


*html_render.py*:66: AttributeError

*===================== 2 failed, 12 passed in 0.10 seconds
======================*

Jays-MacBook-Pro:lesson7 jayjones$

**Comment by natasha-aleksandrova** [comment]
The test failures you are getting are because you are using `self._open_tag()` function but there is no `def _open_tag` function definition anywhere.

**Pull request closed without merging**